castai/resource_commitments.go

```diff
@@ -0,0 +1,541 @@
+package castai
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/mitchellh/mapstructure"
+	"github.com/samber/lo"
+
+	"github.com/castai/terraform-provider-castai/castai/sdk"
+)
+
+const (
+	fieldCommitmentsAzureReservationsCSV = "azure_reservations_csv"
+	fieldCommitmentsGCPCUDsJSON          = "gcp_cuds_json"
+
+	fieldCommitmentsAzureReservations = "azure_reservations"
+	fieldCommitmentsGCPCUDs           = "gcp_cuds"
+	fieldCommitmentsGCPCUDConfigs     = "gcp_cud_configs"
+)
+
+func resourceCommitments() *schema.Resource {
+	return &schema.Resource{
+		ReadContext:   resourceCastaiCommitmentsRead,
+		CreateContext: resourceCastaiCommitmentsCreate,
+		UpdateContext: resourceCastaiCommitmentsUpdate,
+		DeleteContext: resourceCastaiCommitmentsDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: commitmentsStateImporter,
```

Comment:
There is a mechanism in sdk for simple cases when import can reuse read function.
```suggestion
			StateContext: schema.ImportStatePassthroughContext,
```

Reply:
Good point, I didn't know about this one